### PR TITLE
FEAT : Authentication

### PR DESCRIPTION
**Pull request by MiheerTamkhane**
- Login user
- Signup new user
- Logout

### CODE REVIEW

**Review by prathmeshjagtap — APPROVED** [review]
UI is awesome
Functionality is also working properly
Code is  also clean

**Review by manojasarsa — APPROVED** [review]
Functionality is working as expected.
Code seems clean.

File: src/frontend/features/authentication/authSlice.js [hunk 1]
```diff
@@ -27,7 +28,8 @@ const signupThunk = createAsyncThunk(
       const { data } = await axios.post("/api/auth/signup", { ...userDeatils });
       return data;
     } catch (err) {
-      console.error(err);
+      console.error(err.message);
+      rejectWithValue(err);
```
Comment (gopalbbharadva) [inline src/frontend/features/authentication/authSlice.js:30]: Remove consoles.

**Review by gopalbbharadva — APPROVED** [review]
Code is well structured

**Pull request merged**